### PR TITLE
Make it possible to specify different storage classes for different brokers

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

In some cases, it is useful to be able to specify different storage class for different broker nodes. This might be caused for example by the datacenter topology etc. This PR makes it possible using the new `overrides` property in the persistent claim volumes which can override the default storage class for different brokers.

To make this work with the pod and pvc deletion requested through the annotation, I had to refactor this part of code in the `KafkaAssemblyOperator`. Fur this I had to add a new method `deleteAsync` to the `StatefulsetOperator` for deleting the Stateful sets without cascading the deletion. This cannot be implemented into the `reconcile` method without breaking the inheritance etc. So I used separate method. 

I also move the storage related classes in `api` module to separate package to make it easier to find them and use them.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=MirrorMakerST#testMirrorMakerTlsScramSha,MirrorMakerST#testMirrorMaker

File: api/src/main/java/io/strimzi/api/kafka/model/storage/PersistentClaimStorageOverride.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Configures external listener overrides for broker services and advertised addresses
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/storage/PersistentClaimStorageOverride.java]: Wrong comment I guess ;)

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java [hunk 2]
```diff
@@ -1584,7 +1583,8 @@ int getPodIndexFromPvcName(String pvcName)  {
                             result.complete();
                         } else {
                             // The PVC is Bound and resizing is not in progress => We should check if the SC supports resizing and check if size changed
-                            Quantity currentSize = currentPvc.getStatus().getCapacity().get("storage");
+                            //Quantity currentSize = currentPvc.getStatus().getCapacity().get("storage");
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: remove commented line

**Pull request merged**